contracts/mechanisms/LockDrop.sol

```diff
@@ -1,36 +1,92 @@
 import "openzeppelin-zos/contracts/math/SafeMath.sol";
+import "../token/EdgewareERC20.sol";
 
 /**
  * The LockDrop contract locks up funds of depositors
  * with a verifiable receipt/proof that can be used to
  * redeem tokens on a separate, compatible blockchain.
- * THe tokens with be locked for a specified time length.
+ * The tokens with be locked for a specified time length.
+ * 
+ * The tokens are minted at a rate that grows according
+ * to the chosen locking schedule.
  */
 contract LockDrop {
-    uint public lockDropCapacity;
-    uint public lockDropBeginning;
-    uint public lockDropEnding;
+    uint public capacity;
+    uint public beginning;
+    uint public ending;
+    uint public maxLength;
 
-    mapping (address => uint) lockDropBalances;
+    struct Deposit {
+        uint amount;
+        uint lockEnding;
+    }
+    
+    mapping (address => Deposit[]) deposits;
 
-    constructor(uint length, uint capacity) {
-        lockDropCapacity = capacity;
-        lockDropBeginning = now;
-        lockDropEnding = now * length * 1 day;
+    constructor(uint _lockPeriod, uint _maxLength, uint _capacity) {
+        beginning = now;
+        ending = now + (1 days * _lockPeriod);
+        capacity = _capacity;
+        maxLength = _maxLength;
     }
 
-    function lock() payable public {
-        require(now <= lockDropEnding);
-        uint balance = lockDropBalances[msg.sender]
-        lockDropBalances[msg.sender] = SafeMath.add(balance, msg.value);
+    /**
+     * @dev        Lock function for participating in the lock drop
+     * @param      _length  The length of time chosen for locking ether
+     */
+    function lock(uint _length) payable public {
+        require(now <= ending);
+
+        // Create deposit with paid amount and specified length
+        // of time. The lock ending is determined as the specified
+        // time length after the ending of the lock up period in days.
+        Deposit d = Deposit({
+            amount: msg.value,
+            lockEnding: ending + (_length * 1 days)
+        });
+
+        // Push new deposit
+        deposits[msg.sender].push(d);
+
+        // TODO: Emit Deposit event
     }
     
-    function withdraw() public {
-        require(now > lockDropEnding);
-        require(lockDropBalances[msg.sender] > 0);
-        msg.sender.transfer(lockDropBalances[msg.sender]);
+    /**
+     * @dev        Unlock function reverts the lock before the lock starts
+     * @param      _depositIndex  The deposit index to unlock
+     */
+    function unlock(uint _depositIndex) public {
+        require(now <= ending);
+        require(deposits[msg.sender].length > _depositIndex);
+
+        uint memory amount = deposits[msg.sender][_depositIndex].amount;
+        delete deposits[msg.sender][_depositIndex];
+
+        msg.sender.transfer(amount);
     }
     
+
+    /**
+     * @dev        Withdraw function should withdraw all valid ether after lock
+     */
+    function withdraw() public {
```

Comment:
Since we're distributing X% of the initial tokens in this lockdrop. We need to know 1) how many tokens are locked up 2) and what any "longer lockup"/rewards need to be given. From this, we can calculate how many EdgewareERC20 to let an address withdraw.

We've already calculated the `effectiveAmount` of tokens for a specific `Deposit` and can reference `TotalEffectiveDeposits`. 

Using these we can calculate how many EdgewareERC20 tokens to send to msg.sender.